src/Knp/Menu/Renderer/ListRenderer.php

```diff
@@ -194,16 +201,27 @@ protected function renderLink(ItemInterface $item, array $options = []): string
         return $this->format($text, 'link', $item->getLevel(), $options);
     }
 
+    /**
+     * @param array<string, mixed> $options
+     */
     protected function renderLinkElement(ItemInterface $item, array $options): string
     {
-        return \sprintf('<a href="%s"%s>%s</a>', $this->escape($item->getUri()), $this->renderHtmlAttributes($item->getLinkAttributes()), $this->renderLabel($item, $options));
+        $uri = null !== $item->getUri() ? $this->escape($item->getUri()) : '';
```

Comment:
the uri is never null in that method (otherwise, that would render the wrong HTML anyway).

The proper fix to let phpstan know about it is to use `assert(null !== $item->getUri());` instead.

Reply:
But we're actually allowing `$item->getUri()` to return a `null` value (and `$uri` property to be `null`)

Reply:
in the item, yes. But when it returns `null`, we don't call `renderLinkElement` but `renderSpanElement` instead. So, in that method, it won't ever be null.